source/javascripts/components/ProductTour/ProductTooltip.tsx

```diff
@@ -0,0 +1,185 @@
+import React from "react";
+import { useEffect, useReducer } from "react";
+
+import { Text, Button, Flex, Icon, Link } from "@bitrise/bitkit";
+
+import "./ProductTooltip.scss";
+
+import { State, Action, Tips } from "./types";
+
+interface ProductTooltipProps {
+	tips: Tips[];
+	rect: DOMRect | null;
+	onClose: () => void;
+	onChange: (id: string) => void;
+	onButtonClick: (data?: string) => void;
+}
+
+const initialState: State = {
+	items: [],
+	finished: false
+};
+
+function reducer(state: State, { type, payload }: Action): State {
+	switch (type) {
+		case "reset": {
+			return initialState;
+		}
+		case "init": {
+			const selectedId = (payload as Tips[])[0].id;
+			const selectedIndex = 0;
+			const finished = selectedIndex === (payload as Tips[]).length - 2;
+			return { selectedId, selectedIndex, items: payload as Tips[], finished };
+		}
+
+		case "select": {
+			const found = state.items?.find(item => item.id === payload);
+			if (found) {
+				const selectedIndex = state.items.indexOf(found);
+				const finished = selectedIndex === state.items.length - 1;
+				return { ...state, selectedId: found.id, selectedIndex, finished };
+			}
+			break;
+		}
+		case "next": {
+			const { selectedIndex, items } = state;
+			if (selectedIndex !== undefined) {
+				const nextIndex = selectedIndex + 1;
+				const found = items[nextIndex];
+				if (found) {
+					const finished = selectedIndex === items.length - 2;
+					return {
+						...state,
+						selectedId: found.id,
+						selectedIndex: nextIndex,
+						finished
+					};
+				}
+			}
+			break;
+		}
+		case "prev": {
+			const { selectedIndex, items } = state;
+			if (selectedIndex !== undefined) {
+				const prevIndex = Math.max(selectedIndex - 1, 0);
```

Comment:
I think this can be simplified a bit
```suggestion
			if (selectedIndex) {
				const prevIndex = selectedIndex - 1;
```